google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -743,7 +743,7 @@ private final class SyntaxDecoders {
         b -> codecs.bucketInfo().decode(b).asBucket(GrpcStorageImpl.this);
   }
 
-  private static class TransformingPageDecorator<
+  protected static class TransformingPageDecorator<
```

Comment:
We can drop protected all together, along with making this class final since we don't intend for it to ever be extended.